test/functional/editions_controller_test.rb

```diff
@@ -520,84 +520,167 @@ class EditionsControllerTest < ActionController::TestCase
   end
 
   context "#send_to_publish" do
-    setup do
-      @requester = FactoryBot.create(:user, :govuk_editor, name: "Stub Requester")
-      @edition = FactoryBot.create(:edition, state: "scheduled_for_publishing", publish_at: Time.zone.now + 1.hour)
-      login_as(@requester)
-    end
-
-    context "user has govuk_editor permission" do
+    context "edition is in 'scheduled_for_publishing' state" do
       setup do
-        ScheduledPublisher.stubs(:cancel_scheduled_publishing)
-        PublishWorker.stubs(:perform_async)
+        @requester = FactoryBot.create(:user, :govuk_editor, name: "Stub Requester")
+        @edition = FactoryBot.create(:edition, state: "scheduled_for_publishing", publish_at: Time.zone.now + 1.hour)
+        login_as(@requester)
       end
 
-      should "update the edition status to 'published' and save the comment" do
-        post :send_to_publish, params: {
-          id: @edition.id,
-          comment: "Publishing with immediate effect",
-        }
+      context "user has govuk_editor permission" do
+        setup do
+          ScheduledPublisher.stubs(:cancel_scheduled_publishing)
+          PublishWorker.stubs(:perform_async)
+        end
 
-        assert_equal "Published", flash[:success]
-        @edition.reload
-        assert_equal "published", @edition.state
-        assert_equal "Publishing with immediate effect", @edition.latest_status_action.comment
-        assert_equal @requester.id, @edition.latest_status_action.requester_id
+        should "update the edition status to 'published' and save the comment" do
+          post :send_to_publish, params: {
+            id: @edition.id,
+            comment: "Publishing with immediate effect",
+          }
+
+          assert_equal "Published", flash[:success]
+          @edition.reload
+          assert_equal "published", @edition.state
+          assert_equal "Publishing with immediate effect", @edition.latest_status_action.comment
+          assert_equal @requester.id, @edition.latest_status_action.requester_id
+        end
+
+        should "not update the edition state and render 'send_to_publish' template with an error when an error occurs" do
+          EditionProgressor.any_instance.expects(:progress).returns(false)
+
+          post :send_to_publish, params: {
+            id: @edition.id,
+          }
+
+          assert_template "secondary_nav_tabs/send_to_publish_page"
+          assert_equal "Due to a service problem, the request could not be made", flash[:danger]
+          @edition.reload
```

Comment:
Is `@edition.reload` needed here? I think the state can be tested without that.

Reply:
You're correct, it doesn't seem to require it, we have just followed suit in the test suite, as other tests present for editions in **scheduled_for_publishing** state utilise it in the same way, even though they similarly don't fail if removed. So we did the same in our copy of those tests. (Like on [this line](https://github.com/alphagov/publisher/blob/main/test/functional/editions_controller_test.rb#L672))

Reply:
Yes, I think leaving it for consistency is fine. 👍